ddtrace/debugging/_uploader.py

```diff
@@ -76,64 +93,61 @@ def __init__(self, interval: Optional[float] = None) -> None:
+            "Signal uploader initialized (url: %s, endpoints: %s, interval: %f)",
             di_config._intake_url,
             {t: ut.endpoint for t, ut in self._tracks.items()},
             self.interval,
         )
 
         self._flush_full = False
 
-    def set_track_endpoints(self) -> None:
-        if self._agent_endpoints_cache.trickling():
-            return
-
-        try:
-            agent_info = agent.info()
-            self._agent_endpoints = set(agent_info.get("endpoints", [])) if agent_info is not None else set()
-        except Exception:
-            pass  # nosec B110
-        finally:
-            self._agent_endpoints_cache.turn()
-
-        snapshot_track = "/debugger/v1/input"
-        if "/debugger/v2/input" in self._agent_endpoints:
-            snapshot_track = "/debugger/v2/input"
-        elif "/debugger/v1/diagnostics" in self._agent_endpoints:
-            snapshot_track = "/debugger/v1/diagnostics"
-
-        endpoint_suffix = f"?ddtags={quote(di_config.tags)}" if di_config._tags_in_qs and di_config.tags else ""
-
-        # Only create the tracks if they don't exist to preserve the track queue metadata.
-        if not self._tracks:
-            self._tracks = {
-                SignalTrack.LOGS: UploaderTrack(
-                    endpoint=f"/debugger/v1/input{endpoint_suffix}",
-                    queue=self.__queue__(
-                        encoder=LogSignalJsonEncoder(di_config.service_name), on_full=self._on_buffer_full
-                    ),
-                ),
-                SignalTrack.SNAPSHOT: UploaderTrack(
-                    endpoint=f"{snapshot_track}{endpoint_suffix}",
-                    queue=self.__queue__(
-                        encoder=SnapshotJsonEncoder(di_config.service_name), on_full=self._on_buffer_full
-                    ),
-                ),
-            }
+    def info_check(self, agent_info: Optional[dict]) -> bool:
+        if agent_info is None:
+            # Agent is unreachable
+            return False
+
+        if "endpoints" not in agent_info:
+            # Agent not supported
+            log.debug("Unsupported Datadog agent detected. Please upgrade to 7.49.")
+            return False
+
+        endpoints = set(agent_info.get("endpoints", []))
+        snapshot_track = self._tracks[SignalTrack.SNAPSHOT]
+        snapshot_track.enabled = True
+
+        if "/debugger/v2/input" in endpoints:
+            log.debug("Detected /debugger/v2/input endpoint")
+            snapshot_track.endpoint = f"/debugger/v2/input{self._endpoint_suffix}"
+        elif "/debugger/v1/diagnostics" in endpoints:
+            log.debug("Detected /debugger/v1/diagnostics endpoint fallback")
+            snapshot_track.endpoint = f"/debugger/v1/diagnostics{self._endpoint_suffix}"
         else:
-            self._tracks[SignalTrack.SNAPSHOT].endpoint = f"{snapshot_track}{endpoint_suffix}"
+            snapshot_track.enabled = False
+            log.warning(
```

Comment:
What's the frequency that this warning will be logged? I think we want it to happen at most once right?